docs/introduction/renku.rst

```diff
@@ -0,0 +1,134 @@
+base upon which everything else is built.
+
+We rely on the currently most widespread version control system, `git <https
+://git-scm.com/>`_. If you are unfamiliar with ``git`` it would not hurt to
+read at least some of their `excellent tutorials <https://git-
+scm.com/docs/gittutorial>`_. In Renku we try to take care of most of the
+boiler plate ``git`` commands for you, but you should still be aware that it
+is being used under the hood.
+
+The added benefit of using a version control system like ``git`` is that it also
+automatically encourages you to be creative, explore new ideas, and break things.
+"Branching" is extremely light-weight in ``git`` and allows you to freely
+experiment with complete peace-of-mind that you can always simply restore your
+last sane version of your work if everything happens to go off the rails. This
+is a fantastic advantage in data science, where experimentation is
+a critical part of the discovery process.
+
+Note that in Renku we make use of `git LFS <https://git-lfs.github.com>`_ which
+allows to keep not only the code but also the data related to an analysis under
+version control while keeping the git repository itself small.
+
+Git is used automatically by ``renku``:
+
+* when a ``renku`` command is run, a commit is created with the files that were
+  added or changed during the command execution
+
+* this commit also includes some hidden ``.renku`` metadata that holds the
+  dataset metadata or describes the workflow
+
+* the commit message contains the command you executed, so you can check ``git
+  log`` to see what you did in the past (running a workflow, creating a dataset,
+  initializing a project)
+
+
+External storage (git-LFS by default)
+-------------------------------------
+
+Git is an excellent version control system for text-based files, but is not
+ideal for binary data. Therefore, the usual solution is to keep the data out
+of the repository somehow while still preserving links to it for versioning
+purposes.
+
+By default, Renku projects use `git-LFS <https://git-lfs.github.com/>`_ for
+handling data. If you use ``renku`` commands, most of the data handling is done
+for you. For example:
+
+* when you add data with ``renku dataset`` commands
+
+* when you call ``renku run`` to generate output data
+
+Whichever files are flagged for storing with git-LFS, they are automatically
+separated from other repository files when you push to the server. They can
+be retrieved again from the external (LFS) storage when needed if the
+repository is cloned elsewhere.
+
+Keeping large files in LFS gives users the ability to control the amount of
+local space used by a project; LFS files can be left as pointers, and take up
+virtually no space, or can be pulled if needed.
+
+Renku provides a convenience command ``renku storage pull`` for retrieving data
+from LFS. Similarly, any ``renku`` command (e.g. ``renku run``) will check whether
```

Comment:
` will check whether the data it needs is stored in LFS, and if so, it will preemptively fetch it.